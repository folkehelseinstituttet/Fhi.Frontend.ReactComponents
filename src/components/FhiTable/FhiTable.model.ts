import React from 'react';

export type TableStructure = {
  key: number,
  link?: string,
  selected?: boolean,
  data: React.ReactNode[]
};
