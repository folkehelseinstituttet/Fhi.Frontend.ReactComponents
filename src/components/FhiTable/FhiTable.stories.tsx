/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/jsx-props-no-spreading */

import { ComponentMeta, ComponentStory } from '@storybook/react';
import FhiTable from './FhiTable';

export default {
  title: 'Fhi/Table',
  component: FhiTable,
  argTypes: { onNavigate: { action: 'clicked' } },
} as ComponentMeta<typeof FhiTable>;

const Template: ComponentStory<typeof FhiTable> = (args) => <FhiTable {...args} />;

const defaultArgs = {
  headers: ['Header1', 'Header2'],
  data: [
    {
      key: 1,
      data: ['Testdata1 linje1', 'Testdata2 linje1'],
    },
    {
      key: 2,
      data: ['Testdata1 linje2', 'Testdata2 linje2'],
    },
    {
      key: 3,
      link: '/Link/to/3',
      data: ['Clicky rad felt 1', 'Clicky rad felt 2'],
    },
    {
      key: 4,
      link: '/Link/to/4',
      data: ['Clicky rad felt 1', 'Clicky rad felt 2'],
    },
  ],
};

export const Normal = Template.bind({});
Normal.args = {
  ...defaultArgs,
};
