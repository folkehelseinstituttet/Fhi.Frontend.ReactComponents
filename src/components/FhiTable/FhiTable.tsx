import { FC } from 'react';
import styled from 'styled-components';

import { TableStructure } from './FhiTable.model';

type Props = {
  headers: string[],
  data: TableStructure[],
  onNavigate: (link: string)=>{},
};

const TabellRad = styled.tr`
    &:nth-child(even) {
    background: ${({ theme }) => theme.fhiTableSecondary}; 
  }
`;

const KlikkbarTabellRad = styled(TabellRad)`
  cursor: pointer;
  &:hover {
    background-color: ${({ theme }) => theme.fhiBlueGrey2};
  }
`;

const FhiTable: FC<Props> = ({ data, headers, onNavigate }) => (
  <table className="table table-borderless w-100">
    <thead>
      <tr>
        {headers.map((t) => (
          <th key={t}>{t}</th>
        ))}
      </tr>
    </thead>
    <tbody>
      {data.map((d) => (
        d.link
          ? (
            <KlikkbarTabellRad
              key={d.key}
              onClick={() => {
                if (d.link) {
                  onNavigate(d.link);
                }
              }}
            >
              {d.data.map((v, i) => (
                // eslint-disable-next-line react/no-array-index-key
                <td key={i}>{v}</td>
              ))}
            </KlikkbarTabellRad>
          ) : (
            <TabellRad key={d.key}>
              {d.data.map((v, i) => (
                // eslint-disable-next-line react/no-array-index-key
                <td key={i}>{v}</td>
              ))}
            </TabellRad>
          )
      ))}
    </tbody>
  </table>
);

export default styled(FhiTable)``;
