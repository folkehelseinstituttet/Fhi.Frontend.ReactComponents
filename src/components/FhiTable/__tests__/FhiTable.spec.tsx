import { render } from '@testing-library/react';
import Table from '../FhiTable';
import { TableStructure } from '../FhiTable.model';

const testHeaders: string[] = ['Header1', 'Header2'];
const testData: TableStructure[] = [
  {
    key: 1,
    data: ['Testdata11', 'Testdata12'],
  },
  {
    key: 2,
    data: ['Testdata21', 'Testdata22'],
  },
];

describe('FhiDetails', () => {
  test('snapshot test', () => {
    const { container } = render(
      <Table
        className="test"
        data={testData}
        headers={testHeaders}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
