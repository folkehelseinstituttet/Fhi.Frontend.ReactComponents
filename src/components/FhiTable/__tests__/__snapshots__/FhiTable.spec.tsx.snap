// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FhiDetails snapshot test 1`] = `
<table
  className="table table-borderless w-100"
>
  <thead>
    <tr>
      <th>
        Header1
      </th>
      <th>
        Header2
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="FhiTable__TabellRad-sc-wflrh6-0 iliSsu"
    >
      <td>
        Testdata11
      </td>
      <td>
        Testdata12
      </td>
    </tr>
    <tr
      className="FhiTable__TabellRad-sc-wflrh6-0 iliSsu"
    >
      <td>
        Testdata21
      </td>
      <td>
        Testdata22
      </td>
    </tr>
  </tbody>
</table>
`;
